docs/sandbox-permissions.rst

```diff
@@ -1,40 +1,94 @@
+
+Most applications will need access to some of these resources in order to be useful. This is primarily done during the finishing build stage, which can be configured through the ``finish-args`` section of the manifest file (see :doc:`manifests`).
+
+Portals
+-------
+
+Portals have already been mentioned in the :doc:`introduction`. They are a framework for providing access to resources outside of the sandbox, including:
+
+- Opening files with a native file chooser dialog
+- Opening URIs
+- Printing
+- Showing notifications
+- Taking screenshots
+- Inhibiting the user session from ending, suspending, idling or getting switched away
+- Getting network status information
+
+In many cases, portals use a system component to implicitly ask the user for permission before granting access to a particular resource. For example, in the case of opening a file, the user's selection of a file using the file chooser dialog is interpreted as implicitly granting the application access to whatever file is chosen.
+
+This approach enables applications to avoid having to configure blanket access to large amounts of data or services and gives users control over what their applications have access to.
+
+Interface toolkits like GTK and Qt implement transparent support for portals, meaning that applications don't need to any additional work to use them (it is worth checking which portals each toolkit supports). Applications that aren't using a toolkit with support for portals can refer to the `xdg-desktop-portal API documentation <https://flatpak.github.io/xdg-desktop-portal/portal-docs.html>`_ for information on how to use them.
+
+Permissions guidelines
+----------------------
+
+While application developers have control over the sandbox permissions they wish to configure, good practice is encouraged and can be enforced. For example, the Flathub hosting service places requirements on which permissions can be used, and software on the host may warn users if certain permissions are used.
+
+The following guidelines describe which some permissions can be freely used, which can be used on an as-needed basis, and which should be avoided.
+
+Standard permissions
+````````````````````
+
+The following permissions provide access to basic resources that applications commonly require, and can therefore be freely used:
+
+- ``--share=network`` - access the network
+- ``--socket=x11`` - show windows using X11
+- ``--share=ipc`` - share IPC namespace with the host (necessary for X11)
+- ``--socket=wayland`` - show windows with Wayland
+- ``--device=dri`` - OpenGL rendering
+- ``--socket=pulseaudio`` - play sound with PulseAudio
+
+D-Bus access
+````````````
+
+Applications are automatically granted access to their own namespace. Access beyond this is typically unnecessary, although there are a small number of exceptions, such as using `MPRIS to provide media controls <https://www.freedesktop.org/wiki/Specifications/mpris-spec/>`_.
```

Comment:
There is a difference between ownership and talk permissions that should be more clear.

Reply:
I've had a go at fixing this.